src/librustc/ty/sty.rs

```diff
@@ -1848,6 +1848,78 @@ impl<'a, 'gcx, 'tcx> TyS<'tcx> {
             _ => bug!("cannot convert type `{:?}` to a closure kind", self),
         }
     }
+
+    //
+    pub fn has_specified_layout(&self, tcx: TyCtxt<'a, 'gcx, 'tcx>) -> Option<bool> {
```

Comment:
IMO this should go in `intrinsicck` and preferably try to resolve projections, like layout code does.